### PR TITLE
Generate Stream of Simple Payment Transactions

### LINKED ISSUE #3: Generate stream of simple payment transactions
We want to generate a stream of simple payment transactions. In order not to run out of funds, we will just send from one account to itself.

- We will use the account created in #1.
- We will read the frequency at which to send transactions from the environment. This will be a decimal number giving the targeted transactions per second.
- We will also read the fee, and the amount to send from the environment (with default amounts of 1 mina for both).
- We will also read the total time for which to run the generator from the environment. If this is not specified, we will run until killed.
- We will reuse the functionality for signing and sending a transaction via `mina-client` from #1.

### PR DESCRIPTION
**Pull request by ejMina226**
Closes #3

### CODE REVIEW

File: bin/main.ml [hunk 1]
```diff
@@ -46,11 +52,31 @@ let create_account =
...
+  let duration =
+    let doc = "duration length" in
+    Arg.(required & opt (some float) None & info [ "duration" ] ~doc)
+  in
   let doc = "Generate load" in
   let info = Cmd.info "generate-load" ~doc in
-  let generate_load = Ok () in
-  Cmd.v info Term.(const generate_load)
+  let generate_load address amount fee freq duration = 
+    Payment_Stream_Generator_inst.generator address ~amount:amount ~fee:fee freq duration in
```
Comment (joaosreis) [inline bin/main.ml:78]: When passing variables with the same name as the labeled arguments, you can use a simplified syntax, like this:

```suggestion
    Payment_Stream_Generator_inst.generator address ~amount ~fee freq duration in
```

File: lib/payment_stream_generator/payment_stream_generator.ml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      let _ = Mina_cli_inst.send_payment ~receiver: address ~sender: address ~fee:fee amount
+      in
+      Async.Deferred.return ()      
+    in
+    let totalNumberOfTimes freq duration = int_of_float @@ duration *. freq
+    in
+    for _ = 0 to totalNumberOfTimes freq duration do
+      Async.Deferred.don't_wait_for @@ minaCliSendPayment address amount fee;
+      Core_unix.sleep @@ int_of_float @@ 1./.freq
+  done
```
Comment (joaosreis) [inline lib/payment_stream_generator/payment_stream_generator.ml:19]: The code in `main.ml` expects the function to return a `(unit, string) Result.t` type. You can return `Ok ()` after the `for`.
```suggestion
  done;
  Ok ()
```

File: lib/payment_stream_generator/payment_stream_generator.ml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+open! Core
+
+module Unix = UnixLabels
+
+module Make(C : Mina_cli.CONFIG)  = struct
+  module Mina_cli_inst = Mina_cli.Make (C)
+
+  let generator address ?(amount = "1") ?(fee = "1") freq duration =
+    let minaCliSendPayment address amount fee: unit Async.Deferred.t =
+      let _ = Mina_cli_inst.send_payment ~receiver: address ~sender: address ~fee:fee amount
```
Comment (joaosreis) [inline lib/payment_stream_generator/payment_stream_generator.ml:10]: Same as in the other comment, you can simplify
```suggestion
      let _ = Mina_cli_inst.send_payment ~receiver: address ~sender: address ~fee amount
```

File: bin/main.ml [hunk 4]
```diff
@@ -46,11 +52,31 @@ let create_account =
...
+    Arg.(required & opt (some string) None & info [ "address" ] ~doc)
+  in
+  let amount =
+    let doc = "amount to be transferred" in
+    Arg.(required & opt (some string) None & info [ "amount" ] ~doc)
+  in
+  let fee =
+    let doc = "fee" in
+    Arg.(required & opt (some string) None & info [ "fee" ] ~doc)
+  in
```
Comment (joaosreis) [inline bin/main.ml:66]: `amount` and `fee` are optional arguments for the `generator` function, so maybe it's best they are also optional arguments for the CLI

Comment (ejMina226) [reply bin/main.ml:66]: Does this just mean removing the `required` in the `Arg.`?

File: bin/main.ml [hunk 5]
```diff
@@ -46,11 +52,31 @@ let create_account =
...
+    let doc = "duration length" in
+    Arg.(required & opt (some float) None & info [ "duration" ] ~doc)
+  in
   let doc = "Generate load" in
   let info = Cmd.info "generate-load" ~doc in
-  let generate_load = Ok () in
-  Cmd.v info Term.(const generate_load)
+  let generate_load address amount fee freq duration = 
+    Payment_Stream_Generator_inst.generator address amount fee freq duration in
+  Cmd.v info Term.(const generate_load $ address $ amount $ fee $ freq $ duration)
```
Comment (joaosreis) [inline bin/main.ml:79]: With no named arguments, you can just use it like this:
```suggestion
  Cmd.v info Term.(const Payment_Stream_Generator_inst.generator $ address $ amount $ fee $ freq $ duration)
```

**Pull request merged**